### PR TITLE
Added RecurrentTabularExplainer

### PR DESCRIPTION
**Pull request by kgullikson88**
This addresses issue #56, and adds a class to handle tabular data when the model is a stateless keras-style recurrent neural network. In that case, the model expects data with shape `(n_samples, n_timesteps, n_features)`. This mostly uses the machinery of the existing `LimeTabularExplainer`, but reshapes data in a few places so that the user doesn't have to worry about reshaping themselves.

### CODE REVIEW

**Comment by marcotcr** [comment]
Can you share a notebook with an example of a model and explanations, so I can play with it a little bit as well?
Thanks : )

**Comment by kgullikson88** [comment]
I'll work on this, but it will take a few days to find a suitable example dataset that isn't proprietary. A lot of the LSTM examples out there are for text data, which LIME handles differently.

**Comment by marcotcr** [comment]
This is cool. Thank you very much!

**Comment by shwinshaker** [comment]
I find that there is no regression mode in this recurrent explainer, can you add that? Thanks!

**Comment by shwinshaker** [comment]
And it seems that `categorical_features` needs to be updated too, so that it can match the columns of the reshaped data.

**Comment by shakti365** [comment]
I trained a Recurrent Neural Network on MNIST data by transforming it to (n_samples=55000, n_timesteps=28, n_features=28) and then used RecurrentTabularExplainer to generate explanations but it doesn't work very well to explain it. Do you think there might be a problem in fitting the flattened data from RecurrentTabularExplainer on linear model because of the Curse of Dimensionality?

**Comment by denfromufa** [comment]
@kgullikson88 @marcotcr can this RNN explainer work with multiple input features, where every input feature is a sequence of equal length?

**Comment by kgullikson88** [comment]
This explainer will run fine with multiple input features. Keep in mind though, that it is going to fit a linear model with `n_features * sequence_length` features. As @shakti365 mentions, that dimensionality can blow up pretty easily and so the explanations might not be that great.

**Comment by denfromufa** [comment]
@kgullikson88 thanks for explanation! I was able to use your RNN explainer. The prediction is quite slow and explanations does explain very well yet.

I'm getting this warning and it does not make much sense:

https://github.com/marcotcr/lime/blob/master/lime/lime_tabular.py#L290

```
                    Prediction probabilties do not sum to 1, and
                    thus does not constitute a probability space.
                    Check that you classifier outputs probabilities
                    (Not log probabilities, or actual class predictions
```

**Comment by kgullikson88** [comment]
Since you say it doesn't make sense, I assume your predict_proba function does in fact give values that sum to 1? If so, then I'm not sure; maybe @marcotcr would be able to help with that.

**Comment by shakti365** [comment]
@kgullikson88 I have been working on an implementation where we don't flatten out the data to `n_features * sequence_length`. Instead I am trying to mask features across all time steps very similar to how `LimeImageExplainer` works. I have tried this on MNIST data and the explanations make lot more sense the current implementation.

**Comment by shakti365** [comment]
@kgullikson88 @marcotcr Would you be willing to review this approach if I submit a PR?

**Comment by kgullikson88** [comment]
I would take a look, but don't have a lot of time to spend so it might take a while to review.

**Comment by marcotcr** [comment]
Same! I'm trying to graduate, so will probably neglect this repo for a couple more months
